packages/server/src/fhir/operations/find.ts

```diff
@@ -0,0 +1,169 @@
+  status: 'active',
+  kind: 'operation',
+  code: 'find',
+  resource: ['Schedule'],
+  system: false,
+  type: true,
+  instance: true,
+  parameter: [
+    { use: 'in', name: 'start', type: 'dateTime', min: 1, max: '1' },
+    { use: 'in', name: 'end', type: 'dateTime', min: 1, max: '1' },
+    { use: 'out', name: 'return', type: 'Bundle', min: 0, max: '1' },
+  ],
+} as const satisfies OperationDefinition;
+
+type FindParameters = {
+  start: string;
+  end: string;
+};
+
+const TimezoneExtensionURI = 'http://hl7.org/fhir/StructureDefinition/timezone';
+function getTimezone(schedule: Schedule): string | undefined {
+  return (schedule.extension ?? []).find((extension) => extension.url === TimezoneExtensionURI)?.valueCode;
+}
+
+/**
+ * Handles HTTP requests for the Schedule $find operation.
+ *
+ * Endpoints:
+ *   [fhir base]/Schedule/[id]/$find
+ *
+ * @param req - The FHIR request.
+ * @returns The FHIR response.
+ */
+export async function scheduleFindHandler(req: FhirRequest): Promise<FhirResponse> {
+  const ctx = getAuthenticatedContext();
+  const params = parseInputParameters<FindParameters>(findOperation, req);
+  const { start, end } = params;
+
+  // Future performance option: parameterize availability search with this
+  // count so we can quit early once we have identified enough slots.
+  const pageSize = DEFAULT_SEARCH_COUNT;
+
+  const interval = { start: new Date(params.start), end: new Date(params.end) };
+
+  if (interval.start >= interval.end) {
+    throw new OperationOutcomeError(badRequest('Invalid search time range'));
+  }
+
+  if (differenceInDays(interval.end, interval.start) > 31) {
+    throw new OperationOutcomeError(badRequest('Search range cannot exceed 31 days'));
+  }
+
+  const [schedule, slots] = await Promise.all([
+    ctx.repo.readResource<Schedule>('Schedule', req.params.id),
+    ctx.repo.searchResources<Slot>({
+      resourceType: 'Slot',
+
+      count: DEFAULT_MAX_SEARCH_COUNT,
+
+      filters: [
+        {
+          code: 'schedule',
+          operator: Operator.EQUALS,
+          value: `Schedule/${req.params.id}`,
+        },
+
+        {
+          code: '_filter',
+          operator: Operator.EQUALS,
+          // Slot starts sometime in interval, OR
+          // Slot ends sometime in interval, OR
+          // Slot time fully contains interval
+          value: `((start ge "${start}" and start le "${end}") or (end ge "${start}" and end le "${end}") or (start lt "${start}" and end gt "${end}"))`,
+        },
+
+        {
+          code: 'status',
+          operator: Operator.EQUALS,
+          value: 'busy,free,busy-unavailable',
+        },
+      ],
+    }),
+  ]);
+
+  // If we filled a full search page of slots, then there may be slots we
+  // didn't fetch that would impact availability. Fail loudly here.
+  if (slots.length === DEFAULT_MAX_SEARCH_COUNT) {
+    throw new OperationOutcomeError(badRequest('Too many slots found in range; try searching with smaller bounds'));
+  }
```

Comment:
This also is a little awkward. I sort of want a helper like `searchResourcesExhaustive()` that could directly throw if there are rows beyond the search limit? Or maybe there's another query method to use that would let us guarantee that we loaded all relevant records?